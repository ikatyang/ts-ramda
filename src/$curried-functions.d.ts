export type CurriedFunction0<R> = () => R;
export interface CurriedFunction1<T1, R> {
    (v1: T1): R;
}
export interface CurriedFunction2<T1, T2, R> {
    (v1: T1): CurriedFunction1<T2, R>;
    <$SEL extends "1">(): (v1: T1) => CurriedFunction1<T2, R>;
    <$SEL extends "11">(): (v1: T1, v2: T2) => R;
    (v1: T1, v2: T2): R;
}
export interface CurriedFunction3<T1, T2, T3, R> {
    (v1: T1): CurriedFunction2<T2, T3, R>;
    (v1: T1, v2: T2): CurriedFunction1<T3, R>;
    <$SEL extends "1">(): (v1: T1) => CurriedFunction2<T2, T3, R>;
    <$SEL extends "11">(): (v1: T1, v2: T2) => CurriedFunction1<T3, R>;
    <$SEL extends "111">(): (v1: T1, v2: T2, v3: T3) => R;
    (v1: T1, v2: T2, v3: T3): R;
}
export interface CurriedFunction4<T1, T2, T3, T4, R> {
    (v1: T1): CurriedFunction3<T2, T3, T4, R>;
    (v1: T1, v2: T2): CurriedFunction2<T3, T4, R>;
    (v1: T1, v2: T2, v3: T3): CurriedFunction1<T4, R>;
    <$SEL extends "1">(): (v1: T1) => CurriedFunction3<T2, T3, T4, R>;
    <$SEL extends "11">(): (v1: T1, v2: T2) => CurriedFunction2<T3, T4, R>;
    <$SEL extends "111">(): (v1: T1, v2: T2, v3: T3) => CurriedFunction1<T4, R>;
    <$SEL extends "1111">(): (v1: T1, v2: T2, v3: T3, v4: T4) => R;
    (v1: T1, v2: T2, v3: T3, v4: T4): R;
}
export interface CurriedFunction5<T1, T2, T3, T4, T5, R> {
    (v1: T1): CurriedFunction4<T2, T3, T4, T5, R>;
    (v1: T1, v2: T2): CurriedFunction3<T3, T4, T5, R>;
    (v1: T1, v2: T2, v3: T3): CurriedFunction2<T4, T5, R>;
    (v1: T1, v2: T2, v3: T3, v4: T4): CurriedFunction1<T5, R>;
    <$SEL extends "1">(): (v1: T1) => CurriedFunction4<T2, T3, T4, T5, R>;
    <$SEL extends "11">(): (v1: T1, v2: T2) => CurriedFunction3<T3, T4, T5, R>;
    <$SEL extends "111">(): (v1: T1, v2: T2, v3: T3) => CurriedFunction2<T4, T5, R>;
    <$SEL extends "1111">(): (v1: T1, v2: T2, v3: T3, v4: T4) => CurriedFunction1<T5, R>;
    <$SEL extends "11111">(): (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5) => R;
    (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5): R;
}
export interface CurriedFunction6<T1, T2, T3, T4, T5, T6, R> {
    (v1: T1): CurriedFunction5<T2, T3, T4, T5, T6, R>;
    (v1: T1, v2: T2): CurriedFunction4<T3, T4, T5, T6, R>;
    (v1: T1, v2: T2, v3: T3): CurriedFunction3<T4, T5, T6, R>;
    (v1: T1, v2: T2, v3: T3, v4: T4): CurriedFunction2<T5, T6, R>;
    (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5): CurriedFunction1<T6, R>;
    <$SEL extends "1">(): (v1: T1) => CurriedFunction5<T2, T3, T4, T5, T6, R>;
    <$SEL extends "11">(): (v1: T1, v2: T2) => CurriedFunction4<T3, T4, T5, T6, R>;
    <$SEL extends "111">(): (v1: T1, v2: T2, v3: T3) => CurriedFunction3<T4, T5, T6, R>;
    <$SEL extends "1111">(): (v1: T1, v2: T2, v3: T3, v4: T4) => CurriedFunction2<T5, T6, R>;
    <$SEL extends "11111">(): (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5) => CurriedFunction1<T6, R>;
    <$SEL extends "111111">(): (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5, v6: T6) => R;
    (v1: T1, v2: T2, v3: T3, v4: T4, v5: T5, v6: T6): R;
}
