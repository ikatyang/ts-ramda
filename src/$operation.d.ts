import { Morphism } from "./$types";
export type DeepPartial<T> = {
    [K in keyof T]?: DeepPartial<T[K]>;
};
export type Evolver<T> = Morphism<T, T> | {
    [K in keyof T]?: Evolver<T[K]>;
};
export type Diff<T extends string, U extends string> = ({
    [P in T]: P;
} & {
    [P in U]: never;
} & {
    [x: string]: never;
})[T];
export type Omit<T, K extends string> = Pick<T, Diff<keyof T, K>>;
export type Same<T extends string, U extends string> = Diff<T | U, Diff<T, U> | Diff<U, T>>;
export type Merge<T, U> = Omit<T, keyof U> & U;
