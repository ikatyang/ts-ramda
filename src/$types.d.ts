export type Morphism<T, U> = (value: T) => U;
export type IndexedListMorphism<T, U> = (value: T, index: number, list: List<T>) => U;
export type IndexedObjectMorphism<T, U, K extends string> = (value: T, index: number, object: Record<K, T>) => U;
export type ListMapper<T, U> = (fn: Morphism<T, U>, list: List<T>) => U[];
export type ObjectMapper<T, U, K extends string> = (fn: Morphism<T, U>, object: Record<K, T>) => Record<K, U>;
export interface List<T> {
    readonly [index: number]: T;
    readonly length: number;
}
export interface Functor<T> {
    map<U>(fn: Morphism<T, U>): Functor<U>;
}
