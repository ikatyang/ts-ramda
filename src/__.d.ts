import { Placeholder } from "./$placeholder";
declare const $: Placeholder;
export = $;
