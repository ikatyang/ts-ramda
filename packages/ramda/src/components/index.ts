export { Component } from './component';
export { Declaration } from './declaration';
export { Group } from './group';
export { Interface } from './interface';
export { Namespace } from './namespace';
export { Plain } from './plain';
export { Reference } from './reference';
export { Type } from './type';
