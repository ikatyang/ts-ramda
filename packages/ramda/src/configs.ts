export const placeholderName = 'Placeholder';
export const placeholderAbbr = 'PH';
export const namespace = 'R';
export const curriedName = 'CurriedFunction';
