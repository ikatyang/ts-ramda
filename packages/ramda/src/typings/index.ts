export { Typing } from './typing';
export { ArrowTyping } from './arrow';
export { FunctionTyping } from './function';
