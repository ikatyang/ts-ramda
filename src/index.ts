export * from './components/index';
export * from './typings/index';
export * from './helpers/index';
export * from './configs';
