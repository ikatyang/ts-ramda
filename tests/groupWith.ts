import * as R_groupWith from 'ramda/src/groupWith';

declare const string_to_boolean: (x: string) => boolean;
declare const string_array: string[];
declare const string: string;

// @dts-jest
R_groupWith(string_to_boolean);
// @dts-jest
R_groupWith(string_to_boolean, string);
// @dts-jest
R_groupWith(string_to_boolean, string_array);
