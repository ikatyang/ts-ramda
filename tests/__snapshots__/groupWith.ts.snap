// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`R_groupWith(string_to_boolean) 1`] = `"<U extends List<string>>(list: U) => U[]"`;

exports[`R_groupWith(string_to_boolean, string) 1`] = `"string[]"`;

exports[`R_groupWith(string_to_boolean, string_array) 1`] = `"string[][]"`;
